File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Calendars/getcalendarexceptions.rb
module Asposetasksjava
  module GetCalendarExceptions
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        calendars = project.getCalendars().toList()

        i = 0
        while i < calendars.size()
          cal = calendars.get(i)
          calexc = cal.getExceptions().getCount()
          if calexc > 0
            puts "From: " + calexc.getFromDate().toString()
            puts "To: " + calexc.getToDate().toString()
          end
          i +=1
        end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithCalendarExceptions/GetCalendarExceptions.php
<?php
namespace Aspose\Tasks\WorkingWithCalendarExceptions;

use com\aspose\tasks\Project as Project;
class GetCalendarExceptions{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');
        $calendars = $project->getCalendars()->toList();

        $i=0;
        while ($i<sizeof($calendars)) {
            $cal = $calendars->get($i);
            $calexc = $cal->getExceptions()->getCount();
          if (sizeof($calexc)>0){
            print "From: " .(string)$calexc->getFromDate().PHP_EOL;// -> toString();
            print "To: " .(string)$calexc->getToDate().PHP_EOL;// -> toString();
          }
          $i += 1;
        }
    }
}
?>